src/AppBundle/Entity/Post.php

```diff
@@ -203,17 +203,16 @@ public function getComments()
 
     public function addComment(Comment $comment)
     {
+        $comment->setPost($this);
         if (!$this->comments->contains($comment)) {
             $this->comments->add($comment);
-            $comment->setPost($this);
         }
     }
 
     public function removeComment(Comment $comment)
     {
-        if ($this->comments->removeElement($comment)) {
```

Comment:
@bocharsky-bw, this code checks that a comment belongs to a post. See http://www.doctrine-project.org/api/common/2.3/class-Doctrine.Common.Collections.Collection.html#_removeElement